src/rust/ide/enso-protocol/src/language_server/types.rs

```diff
@@ -342,6 +336,15 @@ pub struct CapabilityRegistration {
     pub register_options: RegisterOptions
 }
 
+impl CapabilityRegistration {
+    /// Create "text/canEdit" capability for path
+    pub fn create_can_edit(path:Path) -> Option<Self> {
+        let method           = "text/canEdit".to_string();
+        let register_options = RegisterOptions::ReceivesTreeUpdates(ReceivesTreeUpdates {path});
```

Comment:
This is IMO incorrect. `ReceivesTreeUpdates` is the options for `file/receivesTreeUpdates`, not for `text/canEdit`.
The latter should get its own variant in the `RegisterOptions`.

This will likely work as-is, since both structures are identical (and the name is not serialized) but still this is by accident.